lib/operations/collection_ops.js

```diff
@@ -156,14 +156,17 @@ function bulkWrite(coll, operations, options, callback) {
 
 // Check the update operation to ensure it has atomic operators.
 function checkForAtomicOperators(update) {
-  const keys = Object.keys(update);
+  const keys = Array.isArray(update)
+    ? update.reduce((keys, u) => keys.concat(Object.keys(u)), [])
+    : Object.keys(update);
 
   // same errors as the server would give for update doc lacking atomic operators
   if (keys.length === 0) {
     return toError('The update operation document must contain at least one atomic operator.');
   }
 
-  if (keys[0][0] !== '$') {
+  const foundInvalid = keys.some(key => key[0] !== '$');
```

Comment:
The original assertion here, `keys[0][0] !== '$'` is checking if the *first* key in the object  to see if it is an atomic operator. The new assertion, `keys.some(key => key[0] !== '$')`, is checking if *every* key is an atomic operator. Is that what we want to be doing?

We might want to something like:

```js
function checkForAtomicOperators(update) {
  if (Array.isArray(update)) {
    return update.reduce((err, u) => err || checkForAtomicOperators(u), null);
  }

  // keep the rest of the original function
  ...
}
```

Reply:
Now that we have the `return update.reduce`, we should return `foundInvalid` to `keys[0][0] !== '$'`

Reply:
whoops, thanks for catching that one